server/odc-service/src/main/java/com/oceanbase/odc/service/connection/logicaldatabase/LogicalDatabaseChangeService.java

```diff
@@ -0,0 +1,229 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.connection.logicaldatabase;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.jdbc.lock.JdbcLockRegistry;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.core.shared.PreConditions;
+import com.oceanbase.odc.core.shared.constant.ErrorCodes;
+import com.oceanbase.odc.core.shared.exception.ConflictException;
+import com.oceanbase.odc.metadb.connection.logicaldatabase.LogicalDBChangeExecutionUnitEntity;
+import com.oceanbase.odc.metadb.connection.logicaldatabase.LogicalDBExecutionRepository;
+import com.oceanbase.odc.service.connection.database.DatabaseService;
+import com.oceanbase.odc.service.connection.database.model.Database;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.executor.execution.model.ExecutionStatus;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.executor.sql.SqlExecutionResultWrapper;
+import com.oceanbase.odc.service.connection.logicaldatabase.core.model.LogicalDBChangeExecutionUnit;
+import com.oceanbase.odc.service.connection.logicaldatabase.model.SqlExecutionUnitResp;
+import com.oceanbase.odc.service.schedule.ScheduleService;
+import com.oceanbase.odc.service.task.constants.JobParametersKeyConstants;
+import com.oceanbase.odc.service.task.exception.JobException;
+
+import lombok.NonNull;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/9/3 20:09
+ * @Description: []
+ */
+@Service
+public class LogicalDatabaseChangeService {
+    private final LogicalDatabaseExecutionMapper mapper = LogicalDatabaseExecutionMapper.INSTANCE;
+
+    @Autowired
+    private DatabaseService databaseService;
+
+    @Autowired
+    private ScheduleService scheduleService;
+
+    @Autowired
+    private LogicalDBExecutionRepository executionRepository;
+
+    @Autowired
+    private JdbcLockRegistry jdbcLockRegistry;
+
+    public boolean upsert(List<LogicalDBChangeExecutionUnit> executionUnits) throws InterruptedException {
+        PreConditions.notEmpty(executionUnits, "executionUnits");
```

Comment:
if all executionUnits must belong to same scheduleTask, add validation.

better if involve another parameter

Reply:
validation added.